src/software/gui/standalone_simulator/widgets/standalone_simulator_draw_function_visualizer.cpp

```diff
@@ -64,23 +91,48 @@ void StandaloneSimulatorDrawFunctionVisualizer::mouseMoveEvent(QMouseEvent* even
     auto physics_robot = robot.lock();
     if (physics_robot && standalone_simulator)
     {
-        Point point_in_scene = createPoint(mapToScene(event->pos()));
-        physics_robot->setPosition(point_in_scene);
+        if (shift_clicked)
+        {
+            Point point_in_scene = createPoint(mapToScene(event->pos()));
+            physics_robot->setPositionAndOrientation(point_in_scene,
+                                                     physics_robot->orientation());
+        }
+        else if (r_clicked)
+        {
+            final_click_point = createPoint(mapToScene(event->pos()));
+            Vector vec        = Vector(final_click_point.x() - initial_click_point.x(),
+                                final_click_point.y() - initial_click_point.y());
+            Angle angle       = vec.orientation();
+            physics_robot->setPositionAndOrientation(initial_click_point, angle);
```

Comment:
an alternative would be
```suggestion
            physics_robot->setPositionAndOrientation(physics_robot->position(), angle);
```